x/stakeibc/keeper/transfer.go

```diff
@@ -0,0 +1,102 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	ibctypes "github.com/cosmos/ibc-go/v7/modules/apps/transfer/types"
+	clienttypes "github.com/cosmos/ibc-go/v7/modules/core/02-client/types"
+
+	"github.com/Stride-Labs/stride/v14/x/stakeibc/types"
+)
+
+// Transfers tokens from the community pool deposit ICA account to the host zone holding address for that pool
+func (k Keeper) IBCTransferCommunityPoolTokens(ctx sdk.Context, token sdk.Coin, communityPoolHostZone types.HostZone, shouldLiquidStake bool) error {
+
+	// The memo may contain autopilot commands to atomically liquid stake tokens when transfer succeeds
+	//  both transfer+liquid stake will succeed and stTokens will end in the stride side holding address, 
+	//  or neither will and the original base tokens will return to the foreign deposit ICA address
+	memoCommands := ""
+	autopilotCmd := "{ \"autopilot\": { \"receiver\": \"%s\",  \"stakeibc\": { \"action\": \"LiquidStake\" } } }"
```

Comment:
It might make sense to import this from autopilot. It's not quite as clean, but it does protect against the case where we change the schema

```go
import autopilottypes "github.com/Stride-Labs/stride/v14/x/autopilot/types"

autopilotMetadata := autopilottypes.RawPacketMetadata{
    Autopilot: {
       Receiver: communityPoolHostZone.CommunityPoolHoldingAddress,
       Stakeibc: {
         Action: "LiquidStake"
       }
    }
}

autopilotCmdBz, err := json.Marshal(autopilotMetadata)
autopilotCmd := string(autopilotCmdBz)
```

You might need to pull out the AutopilotMetadata struct since I believe it's private.

Reply:
Good point, this is actually how I was trying to do the PFM meta data originally but I couldn't get it to import the right library so I gave up and just did the json route.  But since autopilot is a stride module we don't need to import from an external hosted location I will take your advice here and use the hard types instead of json